packages/contracts/contracts/test/OptimismPortal.t.sol

```diff
@@ -0,0 +1,190 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+/* Testing utilities */
+import { DSTest } from "../../lib/ds-test/src/test.sol";
+import { Vm } from "../../lib/forge-std/src/Vm.sol";
+import { L2OutputOracle_Initializer } from "./L2OutputOracle.t.sol";
+
+/* Target contract dependencies */
+import { L2OutputOracle } from "../L1/L2OutputOracle.sol";
+import { WithdrawalVerifier } from "../L1/Lib_WithdrawalVerifier.sol";
+
+/* Target contract */
+import { OptimismPortal } from "../L1/OptimismPortal.sol";
+
+contract OptimismPortal_finalizeWithdrawalTransaction_Test is DSTest {
+
+    event TransactionDeposited(
+        address indexed from,
+        address indexed to,
+        uint256 mint,
+        uint256 value,
+        uint256 gasLimit,
+        bool isCreation,
+        bytes data
+    );
+
+    // Utilities
+    Vm vm = Vm(HEVM_ADDRESS);
+    bytes32 nonZeroHash = keccak256(abi.encode("NON_ZERO"));
+
+    // Dependencies
+    L2OutputOracle oracle;
+
+    // Oracle constructor arguments
+    address sequencer = 0x000000000000000000000000000000000000AbBa;
+    uint256 submissionInterval = 1800;
+    uint256 l2BlockTime = 2;
+    bytes32 genesisL2Output = keccak256(abi.encode(0));
+    uint256 historicalTotalBlocks = 100;
+
+    // Test target
+    OptimismPortal op;
+
+    // Target constructor arguments
+    address withdrawalsPredeploy = 0x4200000000000000000000000000000000000015;
+
+    // Cache of timestamps
+    uint256 startingBlockTimestamp;
+    uint256 appendedTimestamp;
+
+    // By default the first block has timestamp zero, which will cause underflows in the tests,
+    // so we jump ahead to the exact time that I wrote this line.
+    uint256 initTime = 1648757197;
+
+    // Withdrawal call parameters
+    uint256 wdNonce = 0;
+    address wdSender = 0xDe3829A23DF1479438622a08a116E8Eb3f620BB5;
+    address wdTarget = 0x1111111111111111111111111111111111111111;
+    uint256 wdValue = 0;
+    uint256 wdGasLimit = 50_000;
+    bytes wdData =
+        hex"111111111111111111111111111111111111111111111111111111111111111111111111111111111111";
+
+    // Generate an output that we can work with. We can use whatever values we want
+    // except for the withdrawerStorageRoot. This one was generated by running the withdrawer.spec.ts
+    // test script against Geth.
+    bytes32 version = bytes32(hex"00");
+    bytes32 stateRoot = keccak256(abi.encode(1));
+    bytes32 withdrawerStorageRoot = 0xb8576230d94535779ec872748df80a094fcad002a8fc2b37c5b8fe250b384be6; // eth_getProof (storageHash)
+    bytes32 latestBlockhash = keccak256(abi.encode(2));
+
+    // This proof was generated by running
+    // make devnet-up
+    // Then in another terminal
+    // packages/contracts/test/withdrawer.spec.ts
+    // Invalid large internal hash
+    bytes withdrawalProof =
```

Comment:
I think a more ideal way to do this is to have a simple script and use the ffi cheatcode to create the proof, see https://github.com/gakonst/lootloose/blob/7b639efe97836155a6a6fc626bf1018d4f8b2495/src/test/LootLooseMeta.t.sol#L98

Reply:
I agree this is ugly, but good enough for now.